client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -284,73 +288,66 @@ public void run() {
       // SparkSubmit will take care of that for us.
       String master = conf.get("spark.master");
       Preconditions.checkArgument(master != null, "spark.master is not defined.");
-
-      List<String> argv = Lists.newArrayList();
-
+      //---Modifying argv to be a map
+       HashMap<String,String> argv=new HashMap<String,String>();
```

Comment:
The indentation is wrong here. It's also missing spaces around `=`.

Also, I'm not sure what you're gaining by putting all arguments in a hash map. You're just creating more coupling - instead of just using the methods and constants defined in `SparkLauncher` to identify the options to set, you're now creating a different string representation of each option for no benefit I can see.